visualization/app/codeCharta/ui/artificialIntelligence/ArtificialIntelligenceHelper/riskProfileHelper.ts

```diff
@@ -0,0 +1,48 @@
+import { CodeMapNode } from "../../../codeCharta.model"
+import { Percentiles } from "./artificialIntelligence.metricThresholds"
+import percentRound from "percent-round"
+import { getAssociatedMetricThresholds } from "./artificialIntelligenceCalculationHelper"
+
+export interface RiskProfile {
+	lowRisk: number
+	moderateRisk: number
+	highRisk: number
+	veryHighRisk: number
+}
+
+export const HEIGHT_METRIC = "mcc"
+export const AREA_METRIC = "rloc"
+export const EXCLUDED_FILE_EXTENSION = new Set(["html", "sass", "css", "scss", "txt", "md", "json", undefined])
+
+export function calculateRiskProfile(node: CodeMapNode, rlocRisk: RiskProfile, fileExtension: string) {
+	const languageSpecificThresholds = getAssociatedMetricThresholds(fileExtension)
+	const thresholds = languageSpecificThresholds[HEIGHT_METRIC]
+	const nodeMetricValue = node.attributes[HEIGHT_METRIC]
+	const nodeRlocValue = node.attributes[AREA_METRIC]
+
+	aggregateRlocRisk(nodeMetricValue, thresholds, nodeRlocValue, rlocRisk)
+}
+
+function aggregateRlocRisk(nodeMetricValue: number, thresholds: Percentiles, nodeRlocValue: number, rlocRisk: RiskProfile) {
+	if (nodeMetricValue <= thresholds.percentile70) {
+		return (rlocRisk.lowRisk += nodeRlocValue)
+	}
+	if (nodeMetricValue <= thresholds.percentile80) {
+		return (rlocRisk.moderateRisk += nodeRlocValue)
+	}
+	if (nodeMetricValue <= thresholds.percentile90) {
+		return (rlocRisk.highRisk += nodeRlocValue)
+	}
+	return (rlocRisk.veryHighRisk += nodeRlocValue)
```

Comment:
I know this code is only moved, it would however be good to update the code to signal that the metrics are not meant to be returned and instead manipulate the input object.